src/TerminalPlayer.c

```diff
@@ -49,11 +50,11 @@ int parse_args(int argc, char **argv)
         switch (ret)
         {
         case 'c':
-            if (!strcmp(optarg, "rgb"))
+            if (!strncmp(optarg, "rgb", 3))
             {
                 rgb_flag = 1;
             }
-            else if (!strcmp(optarg, "gray"))
+            else if (!strncmp(optarg, "gray", 4))
```

Comment:
This string comparison will not work correctly because only the first n characters are compared. Any argument that starts this way will be valid even if it is not. Also, using the "safe" version of strcmp (strncmp), which requires the number of characters to compare to be within bounds, does not make sense because any element (inside argc) will end in \0. Also, the size passed to strncmp is incorrect because the string passed (s2) is a null-terminated string.